forge/db/migrations/20230816-01-add-ApplicationId-to-Devices.js

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-unused-vars */
+
+/**
+ * Add ProjectSnapshots table
+ */
+const { DataTypes } = require('sequelize')
+
+module.exports = {
+    /**
+     * upgrade database
+     * @param {QueryInterface} context Sequelize.QueryInterface
+     */
+    up: async (context) => {
+        await context.addColumn('Devices', 'ApplicationId', {
+            type: DataTypes.INTEGER,
+            references: { model: 'Applications', key: 'id' },
+            onDelete: 'set null',
+            onUpdate: 'cascade'
+        })
```

Comment:
Does the migration need to also populate the `ApplicationId` column for all existing devices?

Reply:
No. As per existing devices linked to an instance, the `project` object has its own relationship to Application - this remains.

The presence of this property is to signify a device is lined to an application not a project.

Reply:
Does this mean the `/applications/<id>/devices` endpoint only returns the devices assigned to just the application? I was working on the assumption it returned the superset of devices assigned to the application directly and devices assigned to instances within the application?

Reply:
Apologies if that has been lost in translation. The current implementation is literal i.e. "Allow a Device to be assigned to Application" as per my understanding and interpretation. 

As it happens, I am not totally sold on your counter proposal - if feels odd somehow?  Perhaps with a rejig of the UI that might feel right but in the current form, it more _feels_ right to me. 

And of course, as this all changes again when grouping is introduced, I/you might feel different about one vs other? And anyhow, the design stage should route this out and I will be sure to ask for clarification to minimise confusion.

Reply:
It's a matter of perspective. Instances are owned by an Application. If a Device is assigned to an instance then it is 'owned' by the parent Application. So a view of devices in the application would include those that have been further assigned to individual instances. But regardless, as I said, we can proceed with what you have and I'll write up a proposed set of changes next week we can discuss in the broader context.